### PR TITLE
Remove Libuv transport and dependency 

### LINKED ISSUE #28067: Slightly flaky test: `UnixDomainSocketsTest.TestUnixDomainSocketWithUrl()`
Test failed just once over thirty days according to the test failure report (<https://dev.azure.com/dnceng/public/_test/analytics?definitionId=837&contextType=build>) filtered for `TestUnixDomainSocketWithUrl` but not branch. It hasn't failed at all in aspnetcore-ci except as a result of other test issues e.g. dotnet installation failures on Helix agents.

- [Build log](https://dev.azure.com/dnceng/public/_build/results?buildId=888925)
- [Helix console log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-heads-master-1f9bba12b03c469690/Libuv.FunctionalTests--net6.0/console.58ac5a42.log?sv=2019-07-07)
- [UnixDomainSocketsTest_TestUnixDomainSocketWithUrl.log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-heads-master-1f9bba12b03c469690/Libuv.FunctionalTests--net6.0/UnixDomainSocketsTest_TestUnixDomainSocketWithUrl.log?sv=2019-07-07)

Failure:
``` text
[xUnit.net 00:00:02.87]     TestUnixDomainSocketWithUrl [FAIL]
[xUnit.net 00:00:02.87]       System.ArgumentOutOfRangeException : Length cannot be less than zero. (Parameter 'length')
[xUnit.net 00:00:02.87]       Stack Trace:
[xUnit.net 00:00:02.87]            at System.String.Substring(Int32 startIndex, Int32 length)
[xUnit.net 00:00:02.87]         /_/src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs(191,0): at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.UnixDomainSocketsTest.TestUnixDomainSocketWithUrl()
[xUnit.net 00:00:02.87]         --- End of stack trace from previous location ---
```
Seems like the `httpResponse` string didn't contain the expected two spaces. Unfortunately, the actual string is not logged.

Nothing obvious in UnixDomainSocketsTest_TestUnixDomainSocketWithUrl.log:
``` text
[0.002s] [TestLifetime] [Information] Starting test TestUnixDomainSocketWithUrl at 2020-11-17T12:52:29
[0.022s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.041s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://unix://var/folders/dq/brw46d8x763fv_mtdxg2_f2w0000gs/T/tmpZ1y7z8.tmp
[0.041s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly Libuv.FunctionalTests
[0.041s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.041s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.042s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /private/tmp/helix/working/AFE4095E/w/BE2F0A15/e/
[0.042s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.045s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HM4AUUI4FEE0" received FIN.
[0.045s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM4AUUI4FEE0" accepted.
[0.045s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM4AUUI4FEE0" started.
[0.046s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM4AUUI4FEE0" disconnecting.
[0.046s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HM4AUUI4FEE0" sending FIN because: "The client closed the connection."
[0.046s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM4AUUI4FEE0" stopped.
[0.051s] [TestLifetime] [Information] Finished test TestUnixDomainSocketWithUrl in 0.0484749s
```

### LINKED ISSUE #36931: Flaky test - WriteAfterConnectionCloseNoops-http_127.0.0.1_0
## Failing Test(s)

- Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests.WriteAfterConnectionCloseNoops

## Error Message

```text
System.TimeoutException : The operation has timed out.
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Shared/TaskExtensions.cs:line 143
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests.WriteAfterConnectionCloseNoops(ListenOptions listenOptions) in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 186
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests.WriteAfterConnectionCloseNoops(ListenOptions listenOptions) in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 193
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[0.002s] [TestLifetime] [Information] Starting test WriteAfterConnectionCloseNoops-http_127.0.0.1_0 at 2021-09-24T12:54:51
[0.004s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.006s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://127.0.0.1:64434
[0.006s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly Libuv.FunctionalTests
[0.007s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.007s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.007s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /private/tmp/helix/working/9BD208CB/w/A936094C/e/
[0.007s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.007s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] TestServer is listening on port 64434
[0.009s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HMBVB9VLT33O" received FIN.
[0.010s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMBVB9VLT33O" accepted.
[0.010s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMBVB9VLT33O" started.
[0.010s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HMBVB9VLT33O" sending FIN because: "The client closed the connection."
[0.010s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMBVB9VLT33O" disconnecting.
[0.010s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMBVB9VLT33O" stopped.
[30.154s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopping
[30.154s] [Microsoft.Hosting.Lifetime] [Information] Application is shutting down...
[30.155s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopped
[30.208s] [Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests] [Error] Test threw an exception.
System.TimeoutException: The operation has timed out.
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Shared/TaskExtensions.cs:line 143
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests.WriteAfterConnectionCloseNoops(ListenOptions listenOptions) in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 186
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.ResponseTests.WriteAfterConnectionCloseNoops(ListenOptions listenOptions) in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 193
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 273
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 54
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 96
[30.208s] [TestLifetime] [Information] Finished test WriteAfterConnectionCloseNoops-http_127.0.0.1_0 in 30.2052938s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1383371&view=ms.vss-test-web.build-test-results-tab&runId=40310620&resultId=119125&paneView=debug

### LINKED ISSUE #37990: ServerCanAbortConnectionAfterUnobservedClose-Loopback test failure
## Failing Test(s)

Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests.ServerCanAbortConnectionAfterUnobservedClose(listenOptionsName: "Loopback")

## Error Message

```text
System.TimeoutException : The operation has timed out.
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Shared/TaskExtensions.cs:line 143
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests.ServerCanAbortConnectionAfterUnobservedClose(String listenOptionsName) in /_/src/Servers/Kestrel/test/FunctionalTests/RequestTests.cs:line 804
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests.ServerCanAbortConnectionAfterUnobservedClose(String listenOptionsName) in /_/src/Servers/Kestrel/test/FunctionalTests/RequestTests.cs:line 809
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[0.002s] [TestLifetime] [Information] Starting test ServerCanAbortConnectionAfterUnobservedClose-Loopback at 2021-10-30T12:50:06
[0.003s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.006s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://127.0.0.1:56242
[0.006s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly Libuv.FunctionalTests
[0.006s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.006s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.006s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /private/tmp/helix/working/B13209A2/w/ADC8097D/e/
[0.006s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.006s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] TestServer is listening on port 56242
[0.007s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMCRKF61AP2P" accepted.
[0.007s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HMCRKF61AP2P" received FIN.
[0.007s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMCRKF61AP2P" started.
[0.007s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMCRKF61AP2P" disconnecting.
[0.007s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv] [Debug] Connection id "0HMCRKF61AP2P" sending FIN because: "The client closed the connection."
[0.008s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMCRKF61AP2P" stopped.
[30.016s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopping
[30.016s] [Microsoft.Hosting.Lifetime] [Information] Application is shutting down...
[30.016s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopped
[30.124s] [Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests] [Error] Test threw an exception.
System.TimeoutException: The operation has timed out.
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Shared/TaskExtensions.cs:line 143
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests.ServerCanAbortConnectionAfterUnobservedClose(String listenOptionsName) in /_/src/Servers/Kestrel/test/FunctionalTests/RequestTests.cs:line 804
   at Microsoft.AspNetCore.Server.Kestrel.Libuv.FunctionalTests.RequestTests.ServerCanAbortConnectionAfterUnobservedClose(String listenOptionsName) in /_/src/Servers/Kestrel/test/FunctionalTests/RequestTests.cs:line 809
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 96
[30.124s] [TestLifetime] [Information] Finished test ServerCanAbortConnectionAfterUnobservedClose-Loopback in 30.1215817s
```
</details>

Looks like the hang is because there was no "Connection id ... paused" log and the test uses that to set a TCS.

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1447412&view=ms.vss-test-web.build-test-results-tab&runId=41694470&paneView=debug&resultId=118435

### LINKED ISSUE #38002: Remove Libuv transport
See https://github.com/aspnet/Announcements/issues/424, https://github.com/dotnet/aspnetcore/issues/23409

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #38002
Fixes #28067
Fixes #36931
Fixes #37990

The libuv transport has been obsolete since 5.0 and continues to be a source of test flakyness and maintenance overhead. This change removes the transport, its tests, and its dependencies like libuv.dll.

I did not do any work to consolidate the transport test infrastructure since we're considering adding other kinds of transports in the future.

Announcement: https://github.com/aspnet/Announcements/issues/476, https://github.com/dotnet/aspnetcore/issues/38022

### CODE REVIEW

File: eng/Baseline.xml [hunk 1]
```diff
@@ -71,7 +71,6 @@ Update this list when preparing for a new patch.
   <Package Id="Microsoft.AspNetCore.Owin" Version="5.0.6" />
   <Package Id="Microsoft.AspNetCore.Razor.Language" Version="5.0.6" />
   <Package Id="Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic" Version="5.0.6" />
-  <Package Id="Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv" Version="5.0.6" />
```
Comment (wtgodbe) [inline eng/Baseline.xml:74]: I think this should stay because we did ship the package in 5.0. Won't matter much in a week when we update the baseline to 6.0 anyways, though.

**Review by wtgodbe — APPROVED** [review]

**Review by HaoK — APPROVED** [review]
:shipit:

File: src/Servers/Kestrel/Transport.Sockets/src/AssemblyInfo.cs [hunk 2]
```diff
@@ -12,3 +12,4 @@
...
 [assembly: InternalsVisibleTo("IISExpress.FunctionalTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
 [assembly: InternalsVisibleTo("IIS.FunctionalTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
+[assembly: InternalsVisibleTo("Microsoft.AspNetCore.Server.Kestrel.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (dougbu) [inline src/Servers/Kestrel/Transport.Sockets/src/AssemblyInfo.cs:15]: This line seems unrelated to everything else in this PR. Why is it necessary now❔

File: eng/Baseline.xml (hunk 1 above)
Comment (dougbu) [reply eng/Baseline.xml:74]: Yes, @Tratcher please leave this file and Baseline.Designer.props alone. I'll change the files when updating them to reflect 6.0.0 GA (soon 😺). For now, they should reflect what we shipped in 5.0.x.

**Review by dougbu — CHANGES_REQUESTED** [review]

**Review by halter73 — APPROVED** [review]
😢

File: src/Servers/Kestrel/Transport.Sockets/src/AssemblyInfo.cs (hunk 2 above)
Comment (Tratcher) [reply src/Servers/Kestrel/Transport.Sockets/src/AssemblyInfo.cs:15]: Yeah, this one took me a minute to figure out. The Socket and Libuv projects shared some src files for things like buffer management. The internals of those shared files were being made visible to Kestrel.Tests via InternalsVisibleTo in the Transport.Libuv project. With that removed, I hade to make them visible from the Transport.Sockets project instead.

**Comment by Tratcher** [comment]
Recomendations for handling the code check error?
```
Found changes in 1 API baseline files
error : Detected modification to baseline API files. PublicAPI.Shipped.txt files should only be updated after a major release. See /docs/APIBaselines.md for more information.
error : Modified API baseline files:
error : src/Servers/Kestrel/Transport.Libuv/src/PublicAPI.Shipped.txt
```

**Review by dougbu — APPROVED** [review]
> Recommendations for handling the code check error?

Once the rest of the CI is green, we (@dotnet/aspnet-build) will use our supreme, extra-special, and almost godlike :godmode: powers to ignore the error and merge the changes.

**Comment by Tratcher** [comment]
@dougbu it's ready when you are.

**Comment by davidfowl** [comment]
Obituary:

You served us well. We learned ALOT about interop, threading, and cross platform IO in general because of you. You also inspired optimizations in the current sockets transport in Kestrel. We will remember your contribution to ASP.NET Core, forever.

RIP Libuv.
